lib/gen_stage.ex

```diff
@@ -903,6 +921,12 @@ defmodule GenStage do
     end
   end
 
+  defp noreply_callback(:handle_demand, [demand, state], %{mod: mod, type: :producer_consumer} = stage) do
```

Comment:
You will need to buffer the demand until we get producers. :)

On Sunday, June 19, 2016, Muaaz Saleem <notifications@github.com> wrote:

> In lib/gen_stage.ex
> <https://github.com/elixir-lang/gen_broker/pull/16#discussion_r67615024>:
>
> > @@ -903,6 +921,12 @@ defmodule GenStage do
> >      end
> >    end
> >
> > +  defp noreply_callback(:handle_demand, [demand, state], %{mod: mod, type: :producer_consumer} = stage) do
>
> Got it, the stage must have producers by now.
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/elixir-lang/gen_broker/pull/16/files/71ab11520936768f415832defe4444e1458dd3e9#r67615024>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AAAlboBV7mBC7WuNv9vDHbqwhVEaRhRgks5qNSNPgaJpZM4I5GMX>
> .
>


-- 


*José Valimwww.plataformatec.com.br
<http://www.plataformatec.com.br/>Founder and Director of R&D*